VMware: Provide facility to specify path for template

Akasurde opened the pull request:
##### SUMMARY
When multiple virtual machines or templates are found in environment,
module needs folder path. This fix adds facility to specify folder path
for template or virtual machine if multiples are found.

Fixes: #37087

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
 - Feature Pull Request
 - Bugfix Pull Request

##### COMPONENT NAME
lib/ansible/module_utils/vmware.py
lib/ansible/modules/cloud/vmware/vmware_guest.py

##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes below -->
```
2.6devel
```

Akasurde commented:
cc @cigamit

Akasurde commented:
@pdellaert @tchernomax @dericcrago Could you please review this ?

dericcrago commented:
Hi @Akasurde - just a couple of comments that you can feel free to disregard.
 * If the full path to the template is being passed, does it make more sense to try to use ``FindByInventoryPath``?
 * Do we need tests that include this new functionality / behavior?

pdellaert reviewed (approved):
LGTM

pdellaert commented:
Approving, but a test would be good

trigger305 commented:
Is this going to be merged in as I could use the functionality please?
Thank you.

fconza commented:
Tested on my cluster and it works.

aaronk1 commented:
@Akasurde Any update on this one?

Akasurde committed:
VMware: Provide facility to specify path for template

When multiple virtual machines or templates are found in environment,
module needs folder path. This fix adds facility to specify folder path
for template or virtual machine if multiples are found.

Fixes: #37087

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde commented:
@dericcrago @pdellaert Could you please review this ? vcsim does not support templates so testcase cannot be added. Thanks,

tchernomax reviewed (approved):
LGTM

pdellaert reviewed (approved):
LGTM

Akasurde commented:
rebuild_merge

unkaputtbar112 commented:
Would add, that you need the "/vm/" part after the datacenter to the documentation/example then, took me quite some time, to figure that out.
